PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:11 + add:0, change matches:
-    _inherit = ['account.analytic.applicability']
-    _inherit = ['hr.employee']
-    _inherit = ['account.analytic.line']
-    _inherit = ['ir.ui.menu']
-    _inherit = ['project.collaborator']
-    _inherit = ["project.project"]
-    _inherit = ["project.task"]
-    _inherit = ["project.update"]
-    _inherit = ['res.company']
-    _inherit = ['res.config.settings']
-    _inherit = ['uom.uom']

Total Changes: 24

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_timesheet/models/analytic_applicability.py
+++ b/addons/hr_timesheet/models/analytic_applicability.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class AccountAnalyticApplicability(models.Model):
-    _inherit = ['account.analytic.applicability']
+    _inherit = 'account.analytic.applicability'
     _description = "Analytic Plan's Applicabilities"
 
     business_domain = fields.Selection(

--- a/addons/hr_timesheet/models/hr_employee.py
+++ b/addons/hr_timesheet/models/hr_employee.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     has_timesheet = fields.Boolean(compute='_compute_has_timesheet', groups="hr.group_hr_user,base.group_system", export_string_translation=False)
 

--- a/addons/hr_timesheet/models/hr_timesheet.py
+++ b/addons/hr_timesheet/models/hr_timesheet.py
@@ -12,7 +12,7 @@ from odoo.tools.translate import _
 
 
 class AccountAnalyticLine(models.Model):
-    _inherit = ['account.analytic.line']
+    _inherit = 'account.analytic.line'
 
     def _get_favorite_project_id_domain(self, employee_id=False):
         employee_id = employee_id or self.env.user.employee_id.id

--- a/addons/hr_timesheet/models/ir_http.py
+++ b/addons/hr_timesheet/models/ir_http.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class IrHttp(models.AbstractModel):
-    _inherit = ['ir.http']
+    _inherit = 'ir.http'
 
     def session_info(self):
         """ The widget 'timesheet_uom' needs to know which UoM conversion factor and which javascript

--- a/addons/hr_timesheet/models/ir_ui_menu.py
+++ b/addons/hr_timesheet/models/ir_ui_menu.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class IrUiMenu(models.Model):
-    _inherit = ['ir.ui.menu']
+    _inherit = 'ir.ui.menu'
 
     def _load_menus_blacklist(self):
         res = super()._load_menus_blacklist()

--- a/addons/hr_timesheet/models/project_collaborator.py
+++ b/addons/hr_timesheet/models/project_collaborator.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class ProjectCollaborator(models.Model):
-    _inherit = ['project.collaborator']
+    _inherit = 'project.collaborator'
 
     @api.model
     def _toggle_project_sharing_portal_rules(self, active):

--- a/addons/hr_timesheet/models/project_project.py
+++ b/addons/hr_timesheet/models/project_project.py
@@ -9,7 +9,7 @@ from odoo.tools.translate import _
 
 
 class ProjectProject(models.Model):
-    _inherit = ["project.project"]
+    _inherit = "project.project"
 
     allow_timesheets = fields.Boolean(
         "Timesheets", compute='_compute_allow_timesheets', store=True, readonly=False,

--- a/addons/hr_timesheet/models/project_task.py
+++ b/addons/hr_timesheet/models/project_task.py
@@ -27,7 +27,7 @@ PROJECT_TASK_READABLE_FIELDS = {
 
 
 class ProjectTask(models.Model):
-    _inherit = ["project.task"]
+    _inherit = "project.task"
 
     project_id = fields.Many2one(domain="['|', ('company_id', '=', False), ('company_id', '=?',  company_id), ('is_internal_project', '=', False)]")
     analytic_account_active = fields.Boolean("Active Analytic Account", related='project_id.analytic_account_active', export_string_translation=False)

--- a/addons/hr_timesheet/models/project_update.py
+++ b/addons/hr_timesheet/models/project_update.py
@@ -4,7 +4,7 @@ from odoo import fields, models, api
 
 
 class ProjectUpdate(models.Model):
-    _inherit = ["project.update"]
+    _inherit = "project.update"
 
     display_timesheet_stats = fields.Boolean(compute="_compute_display_timesheet_stats", export_string_translation=False)
     allocated_time = fields.Integer("Allocated Time", readonly=True)

--- a/addons/hr_timesheet/models/res_company.py
+++ b/addons/hr_timesheet/models/res_company.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     @api.model
     def _default_project_time_mode_id(self):

--- a/addons/hr_timesheet/models/res_config_settings.py
+++ b/addons/hr_timesheet/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     module_project_timesheet_holidays = fields.Boolean("Time Off",
         compute="_compute_timesheet_modules", store=True, readonly=False)

--- a/addons/hr_timesheet/models/uom_uom.py
+++ b/addons/hr_timesheet/models/uom_uom.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class UomUom(models.Model):
-    _inherit = ['uom.uom']
+    _inherit = 'uom.uom'
 
     def _unprotected_uom_xml_ids(self):
         # Override
